disassemblers/ofrak_ghidra/ofrak_ghidra/components/ghidra_analyzer.py

```diff
@@ -220,6 +225,51 @@ def _build_ghidra_server_args(self) -> List[str]:
 
         for ghidra_script in self._scripts:
             args.append(ghidra_script)
-            args.append(ghidra_script.split(".")[0].lower())
 
         return args
+
+
+@dataclass
+class GhidraCodeRegionConfig(ComponentConfig):
+    pass
+
+
+class GhidraCodeRegionModifier(Modifier[Optional[GhidraCodeRegionConfig]], OfrakGhidraMixin):
+    id = b"GhidraCodeRegionModifier"
+    targets = (CodeRegion,)
+
+    get_code_regions_script = OfrakGhidraScript(
+        os.path.join(CORE_OFRAK_GHIDRA_SCRIPTS, "GetCodeRegions.java"),
+    )
+
+    async def modify(self, resource: Resource, config: Optional[GhidraCodeRegionConfig] = None):
+        code_region = await resource.view_as(CodeRegion)
+        ghidra_project = await OfrakGhidraMixin.get_ghidra_project(resource)
+
+        ofrak_code_regions = await ghidra_project.resource.get_descendants_as_view(
+            v_type=CodeRegion, r_filter=ResourceFilter(tags=[CodeRegion])
+        )
+
+        backend_code_regions_json = await self.get_code_regions_script.call_script(resource)
+        backend_code_regions = []
+
+        for cr_j in backend_code_regions_json:
+            cr = CodeRegion(cr_j["start"], cr_j["size"])
+            backend_code_regions.append(cr)
+
+        ofrak_code_regions = sorted(ofrak_code_regions, key=lambda cr: cr.virtual_address)
+        backend_code_regions = sorted(backend_code_regions, key=lambda cr: cr.virtual_address)
+
+        for cr in ofrak_code_regions:
```

Comment:
Why do you need to get and iterate all the `ofrak_code_regions` here? It looks like the only reason is to check `if cr.virtual_address == code_region.virtual_address`, but in that case `cr` is `code_region` so there is no need to check for which `cr` matches the vaddr. We already have that code region, it's the variable `code_region`.

Reply:
Good catch. I was using the index in `ofrak_code_regions` of `code_region` hence iterating over them to find it but I changed the inner loop so no need to anymore